JBPM-7894: replace embedded subprocess with reusable subprocess

xiezhang7 opened the pull request.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

xiezhang7 commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/legacy/Bpmn2JsonUnmarshaller.java:
```diff
@@ -5539,6 +5539,13 @@ protected void applyCallActivityProperties(CallActivity callActivity,
             callActivity.setName("");
         }
+
+        if (properties.get("case") != null) {
```
Add metadata to indicate if it is a "CASE".

xiezhang7 commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-backend/src/main/java/org/kie/workbench/common/stunner/cm/backend/indexing/CaseManagementFileIndexer.java:
```diff
@@ -0,0 +1,43 @@
+
+@ApplicationScoped
+@CaseManagementEditor
+public class CaseManagementFileIndexer extends AbstractBpmnFileIndexer {
```
Index CASE for querying.

romartin commented:
ok to test

romartin commented on kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-api/src/main/java/org/kie/workbench/common/services/refactoring/service/ResourceType.java:
```diff
@@ -53,7 +53,14 @@
+    /**
+     * Case Modeler
+     */
+    BPMN_CM("bpmncmid"),
```
tbh no idea for what's this class used for right now... but does not looks good adding specific domain entries (BPMN/CM) here, do we really need this?

xiezhang7 replied:
Hi @romartin ,

This `enum type` is used for indexing and querying (e.g. [BpmnProcessDataEventListener](https://github.com/kiegroup/kie-wb-common/blob/67f08c18873dc41760bc2e0691ceac3f4d542090/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/indexing/BpmnProcessDataEventListener.java#L121) indexes bpmn files, [CalledElementFormProvider](https://github.com/kiegroup/kie-wb-common/blob/67f08c18873dc41760bc2e0691ceac3f4d542090/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/dataproviders/CalledElementFormProvider.java#L60) for processes queries).   

The indexing interfaces use this `enum type` instead of `interface`, that's why I cannot move the entries to CM module.

romartin replied:
Well TBH I don't really know too much the indexation API and how to use it, yet... but adding custom enum entries for BPMN/CM in some module from outside Stunner looks not much portable for me... 
@manstis @pefernan @ederign  any ideas guys?

kiereleaseuser3 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser3 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

kiereleaseuser2 commented:
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

xiezhang7 commented:
Hi @romartin ,

I don't think CM files are indexed. This [line](https://github.com/kiegroup/kie-wb-common/blob/67f08c18873dc41760bc2e0691ceac3f4d542090/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/indexing/BpmnFileIndexer.java#L78) in BpmnFileIndexer checks file extension. It only accepts Bpmn files. CM files are ignored.

Many files are changed because I pulled code of `CalledElementFormProvider`, `BpmnFileIndexer`, `BpmnProcessDataEventListener`, `FindBpmnProcessIdsQuery` to abstract super classes, so that CM can also extend the classes to index CM files. 
Also because of `EmbeddedSubprocess` is replaced, all the classes that have references to `EmbeddedSubprocess` are changed. 
I also created new definition classes `Case`, `CaseReusableSubprocessTaskExecutionSet`, `ProcessReusableSubprocessTaskExecutionSet`, `AbstractReusableSubprocess`, `CaseReusableSubprocess`, `ProcessReusableSubprocess` to support two types of `ReusableSubprocss` with CASE and PROCESS as callable element. The readonly property `Case` in `CaseReusableSubprocessTaskExecutionSet` and `ProcessReusableSubprocessTaskExecutionSet` is used to distinguish the two types of `ReusableSubprocess`.

xiezhang7 commented:
Hi @romartin , 

`Bpmn2JsonUnmarshaller` is the only class that is really changed in bpmn module. The rest changes are refactoring changes for more abstraction. So I only added new test for `Bpmn2JsonUnmarshaller` in bpmn module. Other changes in the module are verified with existing tests. 

For CM api module, I changed definition classes which are simple classes with fields, getters and setters. So I only added new tests in `HashCodeAndEqualityTest` to test these classes.

For CM backend module, tests are created for most of the changes.

For CM client module, classes are only changed to replace `EmbeddedSubprocess`. The corresponding tests are also updated.

hasys commented:
Jenkins retest please

hasys commented:
Jenkins execute full downstream build

romartin commented:
@hasys just as a reminder - this is ready for review. Thanks!!

LuboTerifaj commented:
Hello @xiezhang7 @romartin

I will have a look at it today or tomorrow.

Can you @xiezhang7 please adding me as reviewer for every PR which is related to CM?

Thank you.

xiezhang7 commented:
Sorry @LuboTerifaj , but I don't think I am allowed to change the reviewers of any pull request.

romartin commented:
> Sorry @LuboTerifaj , but I don't think I am allowed to change the reviewers of any pull request.
You cannot assign reviewers sorry, but you can just include @LuboTerifaj in the PR's description and he'll receive notifications, anyway I'll remember to add @LuboTerifaj for the CM reviews from now on

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins retest please this

romartin commented:
sorry @xiezhang7 , after merging you PR about the palette, you'll have to fix the conflicts here now. Thanks!

xiezhang7 commented:
Thanks @romartin , I have resolved the conflicts.

romartin commented:
Jenkins execute full downstream build

hasys commented:
Hi @xiezhang7, @romartin, @LuboTerifaj,

I see that we have failed test with NullPointer: `kie-wb-common/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/indexing/BpmnFileIndexerTest.java`

In one of file created in this PR: https://github.com/kiegroup/kie-wb-common/pull/2291/files#diff-c817231f38b63ece42327ada05f68125R124

To reproduce it just execute `mvn clean test -Dtest=BpmnFileIndexerTest` under `kie-wb-common/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend`

<details>
  <summary>Full stack trace</summary>
  <pre>
org.kie.workbench.common.stunner.bpmn.backend.service.indexing.BpmnFileIndexerTest.testBpmnIndexing

java.lang.NullPointerException
	at org.kie.workbench.common.stunner.bpmn.backend.query.AbstractFindIdsQuery$BpmnProcessIdsResponseBuilder.buildResponse(AbstractFindIdsQuery.java:124)
	at org.kie.workbench.common.services.refactoring.backend.server.query.RefactoringQueryServiceImpl.query(RefactoringQueryServiceImpl.java:214)
	at org.kie.workbench.common.stunner.bpmn.backend.service.indexing.BpmnFileIndexerTest.testBpmnIndexing(BpmnFileIndexerTest.java:219)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Standard Error
[main] INFO org.eclipse.jgit.internal.ketch.KetchSystem - Using 5 threads
[Thread-5] INFO org.uberfire.ext.metadata.io.IndexerDispatcher - Scheduling 1 indexing jobs for cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-6] INFO org.uberfire.ext.metadata.io.IndexerDispatcher - Completed indexing 0 events for indexer [id=org.kie.workbench.common.stunner.bpmn.backend.service.indexing.TestBpmnFileIndexer] in cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-6] INFO org.uberfire.ext.metadata.io.IOServiceIndexedImpl - Completed indexing 1 events in cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-4] INFO org.uberfire.ext.metadata.io.BatchIndex - Starting indexing of testBpmnIndexing/master ...
[Thread-4] INFO org.uberfire.ext.metadata.io.IndexerDispatcher - Scheduling 1 indexing jobs for cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/kie/kie-ci/7.16.0-SNAPSHOT/kie-ci-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.kie.scanner.KieScannerFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.kie.scanner.MavenClassLoaderResolver

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/jbpm/jbpm-flow/7.16.0-SNAPSHOT/jbpm-flow-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.jbpm.marshalling.impl.ProcessMarshallerFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.jbpm.process.instance.ProcessRuntimeFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/kie/kie-dmn-core/7.16.0-SNAPSHOT/kie-dmn-core-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service +org.kie.dmn.core.assembler.DMNAssemblerService

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service +org.kie.dmn.core.runtime.DMNRuntimeService

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service +org.kie.dmn.core.weaver.DMNWeaverService

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/drools/drools-model-compiler/7.16.0-SNAPSHOT/drools-model-compiler-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.drools.modelcompiler.CanonicalKieModuleProvider

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/jbpm/jbpm-flow-builder/7.16.0-SNAPSHOT/jbpm-flow-builder-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.jbpm.process.builder.ProcessBuilderFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/jbpm/jbpm-bpmn2/7.16.0-SNAPSHOT/jbpm-bpmn2-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.jbpm.bpmn2.BPMN2ProcessProviderImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.jbpm.bpmn2.xml.XmlProcessDumperFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/drools/drools-compiler/7.16.0-SNAPSHOT/drools-compiler-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.drools.compiler.kie.builder.impl.KieServicesImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.drools.compiler.builder.impl.KnowledgeBuilderFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/drools/drools-core/7.16.0-SNAPSHOT/drools-core-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.drools.core.io.impl.ResourceFactoryServiceImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.drools.core.marshalling.impl.MarshallerProviderImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.drools.core.concurrent.ExecutorProviderImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Loading kie.conf from  jar:file:/home/jenkins/.m2/repository/org/kie/kie-internal/7.16.0-SNAPSHOT/kie-internal-7.16.0-SNAPSHOT.jar!/META-INF/kie.conf in classloader sun.misc.Launcher$AppClassLoader@6bc7c054
[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.kie.internal.services.KieAssemblersImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.kie.internal.services.KieRuntimesImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.kie.internal.services.KieWeaversImpl

[Thread-6] INFO org.kie.api.internal.utils.ServiceDiscoveryImpl - Adding Service org.kie.internal.services.KieBeliefsImpl

[Thread-4] INFO org.uberfire.ext.metadata.io.IndexerDispatcher - Scheduling 1 indexing jobs for cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-6] INFO org.uberfire.ext.metadata.io.IndexerDispatcher - Completed indexing 1 events for indexer [id=org.kie.workbench.common.stunner.bpmn.backend.service.indexing.TestBpmnFileIndexer] in cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-6] INFO org.uberfire.ext.metadata.io.BatchIndex - Completed indexing of testBpmnIndexing/master
[Thread-5] ERROR org.kie.workbench.common.services.refactoring.backend.server.impact.ResourceReferenceCollector - Existing resource found with type bpmn2id (tried to add resource reference of type bpmn2name)
[Thread-5] ERROR org.drools.core.xml.ExtensibleXmlParser - Process Info: id:multiple-rule-tasks, pkg:org.jbpm, name:multiple-rule-tasks, version:1.0 
Node Info: id: name: 
Parser message: (null: 16, 30): cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements}'. One of '{"http://www.omg.org/spec/BPMN/20100524/MODEL":property, "http://www.omg.org/spec/BPMN/20100524/MODEL":laneSet, "http://www.omg.org/spec/BPMN/20100524/MODEL":flowElement, "http://www.omg.org/spec/BPMN/20100524/MODEL":artifact, "http://www.omg.org/spec/BPMN/20100524/MODEL":resourceRole, "http://www.omg.org/spec/BPMN/20100524/MODEL":correlationSubscription, "http://www.omg.org/spec/BPMN/20100524/MODEL":supports}' is expected.
[Thread-5] ERROR org.drools.core.xml.ExtensibleXmlParser - Process Info: id:multiple-rule-tasks, pkg:org.jbpm, name:multiple-rule-tasks, version:1.0 
Node Info: id: name: 
Parser message: (null: 106, 32): cvc-complex-type.4: Attribute 'scriptFormat' must appear on element 'drools:onEntry-script'.
[Thread-5] ERROR org.drools.core.xml.ExtensibleXmlParser - Process Info: id:multiple-rule-tasks, pkg:org.jbpm, name:multiple-rule-tasks, version:1.0 
Node Info: id: name: 
Parser message: (null: 112, 31): cvc-complex-type.4: Attribute 'scriptFormat' must appear on element 'drools:onExit-script'.
[Thread-5] ERROR org.kie.workbench.common.services.refactoring.backend.server.impact.ResourceReferenceCollector - Existing resource found with type bpmn2id (tried to add resource reference of type bpmn2name)
[Thread-5] ERROR org.drools.core.xml.ExtensibleXmlParser - Process Info: id:multiple-rule-tasks, pkg:org.jbpm, name:multiple-rule-tasks, version:1.0 
Node Info: id: name: 
Parser message: (null: 16, 30): cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements}'. One of '{"http://www.omg.org/spec/BPMN/20100524/MODEL":property, "http://www.omg.org/spec/BPMN/20100524/MODEL":laneSet, "http://www.omg.org/spec/BPMN/20100524/MODEL":flowElement, "http://www.omg.org/spec/BPMN/20100524/MODEL":artifact, "http://www.omg.org/spec/BPMN/20100524/MODEL":resourceRole, "http://www.omg.org/spec/BPMN/20100524/MODEL":correlationSubscription, "http://www.omg.org/spec/BPMN/20100524/MODEL":supports}' is expected.
[Thread-5] ERROR org.drools.core.xml.ExtensibleXmlParser - Process Info: id:multiple-rule-tasks, pkg:org.jbpm, name:multiple-rule-tasks, version:1.0 
Node Info: id: name: 
Parser message: (null: 106, 32): cvc-complex-type.4: Attribute 'scriptFormat' must appear on element 'drools:onEntry-script'.
[Thread-5] ERROR org.drools.core.xml.ExtensibleXmlParser - Process Info: id:multiple-rule-tasks, pkg:org.jbpm, name:multiple-rule-tasks, version:1.0 
Node Info: id: name: 
Parser message: (null: 112, 31): cvc-complex-type.4: Attribute 'scriptFormat' must appear on element 'drools:onExit-script'.
[Thread-5] ERROR org.kie.workbench.common.stunner.bpmn.backend.indexing.AbstractBpmnFileIndexer - Error: Process 'Default Process' [org.jbpm.broken]: Node 'Script Task' [3] Action has invalid action: [Error: unknown class or illegal statement: org.mvel2.ParserContext@70de6c3f]
[Near : {... BAD SCRIPT ....}]
                       ^
[Line: 1, Column: 11].
[Thread-5] ERROR org.kie.workbench.common.stunner.bpmn.backend.indexing.AbstractBpmnFileIndexer - Trying to finish indexing process 'org.jbpm.broken/Default Process' despite 1 validation errors.
[Thread-5] INFO org.uberfire.ext.metadata.io.IndexerDispatcher - Completed indexing 7 events for indexer [id=org.kie.workbench.common.stunner.bpmn.backend.service.indexing.TestBpmnFileIndexer] in cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].
[Thread-5] INFO org.uberfire.ext.metadata.io.IOServiceIndexedImpl - Completed indexing 7 events in cluster [KClusterImpl [clusterId=testBpmnIndexing/master]].</pre>
</details>

LuboTerifaj commented:
Thanks @hasys for the catch.

Can you @xiezhang7 provide the fix?

xiezhang7 commented:
Thanks @hasys @LuboTerifaj , [pull request 2331](https://github.com/kiegroup/kie-wb-common/pull/2331) is created to fix the test.

romartin commented:
OK sorry guys, I saw a failed test on Jenkins but I though it was some of that "already known" tests that were always failing on Jenkins.... my fault. So at this point, you know @LuboTerifaj @hasys  if we fixed all the tests failures that were happening eventually? I mean, should I now always wait for all tests to be passed? Not really sure if that was finally fixed...

Anyway will keep an eye to https://github.com/kiegroup/kie-wb-common/pull/2331 and will merge once the builds are ok

Thanks!

hasys commented:
@romartin so far we have no random fails on Stunner, so if failing test is Stunner/new Designer/CM Designer related it is highly likely that test fail related to current PR. But also it is a probability that we have NEW unstable test so it also better to double check any way.

If test is not related to Stunner, than yes it is more complicated to say what is it.

Thank you!
